src/team_lib/ucx/alltoall/alltoall_pairwise.c

```diff
@@ -256,5 +262,11 @@ xccl_status_t xccl_ucx_alltoall_pairwise_start(xccl_ucx_collreq_t *req)
 
     req->alltoall_pairwise.n_rreqs       = total_reqs;
     req->alltoall_pairwise.n_sreqs       = total_reqs;
+
+    if (req->args.field_mask & XCCL_COLL_OP_ARGS_FIELD_STREAM) {
+        assert(req->src_mem_type != UCS_MEMORY_TYPE_HOST);
```

Comment:
Why such assert? Can't i run collective over HOST buffers but have a cuda stream in my application, and use stream argument ?

Reply:
Do you mean use case where we start collective with host buffers only after some gpu activity done?

Reply:
yes, of vice-versa start collective over host buffer after gpu kernel is done. I know sounds weird and can be resolved at user level. So just asking if we "require" from the user such assert ?

Reply:
you are right, will remove assert